packages/extension-link/src/link.ts

```diff
@@ -322,11 +322,23 @@ export const Link = Mark.create<LinkOptions>({
     return {
       setLink:
         attributes => ({ chain }) => {
+          const { href } = attributes
+
+          if (!isAllowedUri(href, this.options.protocols)) {
+            throw new Error('Invalid protocol')
+          }
```

Comment:
I don't think throwing an error is appropriate here. This makes the caller have to deal with the error.

The more common pattern for commands would be to return false, i.e. I refuse to execute this command

Reply:
Also, we should be calling the user's `isAllowedUri` https://github.com/ueberdosis/tiptap/pull/5945/files#diff-1d1f108bd30f776ec9689dc1bb00a3cb3dafc89fde595994aa8fa400471a87d6R368

Because our `isAllowedUri` does not accept things like custom protocols which some may link to for example: `ftp://, obsidian://` etc.